node/pkg/telemetry/loki.go

```diff
@@ -142,13 +153,35 @@ func NewLokiCloudLogger(ctx context.Context, logger *zap.Logger, url string, pro
 		lokiLabels[level] = levLabels
 	}
 
+	// Create a buffered channel so the application does not block in the logger. The size of this
+	// channel should be big enough to handle a burst of log messages from the app.
+	localChan := make(chan api.Entry, 1000)
+
+	// Create a local context with a cancel function so we can signal our go routine to shutdown when the time comes.
+	localContext, localCancel := context.WithCancel(ctx)
+
+	// Kick off a go routine to read from the local buffered channel and write to the Loki unbuffered channel.
+	go func(ctx context.Context, c client.Client, localC chan api.Entry) {
+		for {
+			select {
+			case <-ctx.Done():
+				c.Stop()
```

Comment:
Will buffered messages in the local channel be processed when this occurs? Or when any error occurs? We might encounter a case where the buffered messages are dropped.

Reply:
Also, I think stop will get called twice now, once here, and again in `close()` which triggered this.

Reply:
This code has changed drastically. Please take another look.